### PR TITLE
refactor(Stream): left/right parameters and generic names

### PR DESCRIPTION
**Pull request by dilame**
Even though it's a draft, feedback is appreciated anyway

### CODE REVIEW

**Comment by tim-smart** [comment]
I think I prefer this. Could you add a changeset?

**Comment by dilame** [comment]
> I think I prefer this. Could you add a changeset?

Glad to hear it! Done 😄

File: .changeset/stream-left-right.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"effect": patch
+---
+
+Introduce new arguments and generics naming style for stream module
```
Comment (tim-smart) [inline .changeset/stream-left-right.md:5]:
```suggestion
Introduce left / right naming for Stream apis
```

File: packages/effect/src/Stream.ts [hunk 2]
```diff
@@ -1034,29 +1034,29 @@ export const concatAll: <A, E, R>(streams: Chunk.Chunk<Stream<A, E, R>>) => Stre
  * @category utils
  */
 export const cross: {
-  <A2, E2, R2>(that: Stream<A2, E2, R2>): <A, E, R>(self: Stream<A, E, R>) => Stream<[A, A2], E2 | E, R2 | R>
-  <A, E, R, A2, E2, R2>(self: Stream<A, E, R>, that: Stream<A2, E2, R2>): Stream<[A, A2], E | E2, R | R2>
+  <AR, EL, RL>(right: Stream<AR, EL, RL>): <AL, ER, RR>(left: Stream<AL, ER, RR>) => Stream<[AL, AR], EL | ER, RL | RR>
```
Comment (tim-smart) [inline packages/effect/src/Stream.ts:1037]:
```suggestion
  <AR, ER, RR>(right: Stream<AR, ER, RR>): <AL, EL, RL>(left: Stream<AL, EL, RL>) => Stream<[AL, AR], EL | ER, RL | RR>
```

File: packages/effect/src/Stream.ts [hunk 3]
```diff
@@ -5519,8 +5519,8 @@ export const zipAllWith: {
  * @category zipping
  */
 export const zipLatest: {
-  <A2, E2, R2>(that: Stream<A2, E2, R2>): <A, E, R>(self: Stream<A, E, R>) => Stream<[A, A2], E2 | E, R2 | R>
-  <A, E, R, A2, E2, R2>(self: Stream<A, E, R>, that: Stream<A2, E2, R2>): Stream<[A, A2], E | E2, R | R2>
+  <AL, EL, RL>(right: Stream<AL, EL, RL>): <AR, ER, RR>(left: Stream<AR, ER, RR>) => Stream<[AL, AR], EL | ER, RL | RR>
```
Comment (tim-smart) [inline packages/effect/src/Stream.ts:5522]: Generics need fixing.

File: packages/effect/src/internal/stream.ts [hunk 4]
```diff
@@ -1680,74 +1680,74 @@ export const concatAll = <A, E, R>(streams: Chunk.Chunk<Stream.Stream<A, E, R>>)
...
-  ) => <A, E, R>(self: Stream.Stream<A, E, R>) => Stream.Stream<[A, A2], E2 | E, R2 | R>,
-  <A, E, R, A2, E2, R2>(
-    self: Stream.Stream<A, E, R>,
-    that: Stream.Stream<A2, E2, R2>
-  ) => Stream.Stream<[A, A2], E2 | E, R2 | R>
->(
+export const cross: {
+  <AR, EL, RL>(
+    right: Stream.Stream<AR, EL, RL>
+  ): <AL, ER, RR>(left: Stream.Stream<AL, ER, RR>) => Stream.Stream<[AL, AR], EL | ER, RL | RR>
```
Comment (tim-smart) [inline packages/effect/src/internal/stream.ts:1686]: Generics need fixing.

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Comment by tim-smart** [comment]
I have changed the base to `main`, so you will likely need to `git reset --hard origin/stream-left-right`

File: packages/effect/src/internal/stream.ts (hunk 4 above)
Comment (dilame) [reply packages/effect/src/internal/stream.ts:1686]: thanks for such a careful review, will do it right now

**Review by tim-smart — APPROVED** [review]

**Pull request merged**